File: frontend/src/app/core/asistente/asistente.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { environment } from '../../../environments/environment';
import { ChatOut, MensajeChat } from './asistente.models';

@Injectable({ providedIn: 'root' })
export class AsistenteService {
  private readonly http = inject(HttpClient);

  /** El backend no persiste nada (CU18): mandamos la conversacion completa en cada request. */
  chatear(mensajes: MensajeChat[]): Observable<ChatOut> {
    return this.http.post<ChatOut>(`${environment.apiUrl}/asistente/chat`, { mensajes });
  }
}

File: movil/lib/core/asistente/asistente_service.dart
import '../api.dart';
import 'asistente_models.dart';

/// CU18: Asistir al Cliente via Chatbot. El backend no persiste nada, asi que
/// mandamos la conversacion completa en cada request (espejo de `AsistenteService` web).
class AsistenteService {
  Future<ChatOut> chatear(List<MensajeChat> mensajes) async {
    final respuesta = await api.post(
      '/asistente/chat',
      cuerpo: {'mensajes': mensajes.map((m) => m.aJson()).toList()},
    );
    return ChatOut.desdeJson(respuesta as Map<String, dynamic>);
  }
}

final asistenteService = AsistenteService();
